cpp/src/arrow/util/memory-pool.cc

```diff
@@ -44,14 +45,22 @@ class InternalMemoryPool : public MemoryPool {
 };
 
 Status InternalMemoryPool::Allocate(int64_t size, uint8_t** out) {
+  constexpr size_t kAlignment = 64;
   std::lock_guard<std::mutex> guard(pool_lock_);
-  *out = static_cast<uint8_t*>(std::malloc(size));
-  if (*out == nullptr) {
+  // TODO(emkornfield) find something compatible with windows
```

Comment:
yep.  I'll do the refactoring but I won't add the windows code, because I don't have anyway of testing it.  I'll open a jira to add windows support via appveyor